lib/rubocop/cop/performance/unneeded_sort.rb

```diff
@@ -3,17 +3,52 @@
 module RuboCop
   module Cop
     module Performance
-      # This cop is used to identify instances of sorting and then taking
-      # only the first or last element.
+      # This cop is used to identify instances of sorting and then
+      # taking only the first or last element. The same behavior can
+      # be accomplished without a relatively expensive sort by using
+      # `Enumerable#min` instead of sorting and taking the first
+      # element and `Enumerable#max` instead of sorting and taking the
+      # last element. Similarly, `Enumerable#min_by` and
+      # `Enumerable#max_by` can replace `Enumerable#sort_by` calls
+      # after which only the first or last element is used.
       #
       # @example
       #   # bad
       #   [].sort.first
-      #   [].sort_by(&:length).last
+      #   [].sort[0]
```

Comment:
You might also put some elements in those brackets or replace them with `arr`, `collection` or something like that. Looks odd to be sorting an empty array literal. ;-)

Reply:
Done! Thanks for all your patience and advice on this!

Reply:
You're welcome!